website/blog/2022-12-13-using-cpts-and-taxonomies.md

```diff
@@ -0,0 +1,61 @@
+slug: using-cpts-and-taxonomies
+authors: obradovic
+date: 2022-12-13
+tags: [eightshift, boilerplate, cpt, custom post type, taxonomy, taxonomies, terms]
+hide_table_of_contents: false
+---
+
+By default, WordPress offers two post types for our content management - Posts and Pages. But sometimes, that is not enough. That's where custom post types and custom taxonomies come in. This blog post will cover the basics of registering CPTs and Taxonomies with Eightshift Dev kit.
+
+<!--truncate-->
+## Why should you use CPTs?
+The great thing about WordPress is how much extendability it offers. With projects becoming more complex, the need for additional post types and taxonomies increases. Sure, you can cram everything into default posts or pages, but this can become very chaotic. Using custom post types allows you to manage your content much better. You can separate galleries and projects from news articles etc.
+
+Taxonomies, like the default category, allow for even more grouping options.
+
+## Registering Custom Post Types
+For this exercise, we'll create a new CPT called `Projects`. To make the process of registering new CPTs as easy as possible, we'll use a WP-CLI command to create our Custom Post Type. To create your new CPT, enter the following command:
+
+```bash
+wp boilerplate create post-type --label='Project' --plural_label='Projects' --slug='project' --rewrite_url='project' --rest_endpoint_slug='projects'
+```
+
+> If your new CPT is not working, try flushing rewrite rules by re-saving the settings in **Settings -> Permalinks** or by using `wp cache flush` CLI command
+
+Your new post type is registered and ready to use! Easy, right?
+
+Sometimes all these parameters can be a bit confusing, so here's a quick reference of best practices when setting these parameters:
+
+| Parameter          | Singular/Plural | Writing style | Example  |
+|--------------------|-----------------|---------------|----------|
+| label              | Singular        | Regular       | Project  |
+| plural_label       | Plural          | Regular       | Projects |
+| slug               | Singular        | kebab-case    | project  |
+| rewrite_url        | Singular        | kebab-case    | project  |
+| rest_endpoint_slug | Plural          | kebab-case    | projects |
+
+## Registering Taxonomies
+Now that we have the new custom post type, we need a way to group the projects. We'll create a custom taxonomy called `Project Category`. As with the CPT registration, the easiest way to register taxonomies is by using the following WP-CLI command:
+
+```bash
+wp boilerplate create taxonomy --label='Project Category' --plural_label='Project Categories' --slug='project-category' --rest_endpoint_slug='project-categories' --post_type_slug='project'
+```
+
+Similar suggestions apply to the parameters when naming taxonomies as well. Be sure to write the correct post type slug for which you are registering this new taxonomy!
+
```

Comment:
I'd also add here a bit about why the naming is better, and how taxonomy and terms are organized in the DB.

Mentioning many to many relationships, etc.

Reply:
I added that part on the bottom, under the `Further Reading` section. WP docs have it explained quite nicely.